### ISSUE TITLE
Handing up during truncate replicated table on cluster 

### ISSUE THREAD

**Issue by bespilotnayakartoshka** [ClickHouse/ClickHouse#6725]
**Describe the bug**
Sometimes during truncate replicated tables whole shard just hanging up. 
There are no any chances to run any queries anymore until timeout is raises.  

**How to reproduce**
1. Create two shards with two replicas each. 
2. Create replicated table on cluster. 
3. Run truncate for this table X times until it lag 

* Which ClickHouse server version to use
19.13, 19.8, 19.6 

* `CREATE TABLE` statements for all tables involved

```
CREATE TABLE test_table ON CLUSTER my_cluster 
(field String) 
ENGINE = ReplicatedMergeTree('/clickhouse/tables/{shard}/test_table', '{replica}')
ORDER BY (field)
```

* Queries to run that lead to unexpected result
```
TRUNCATE TABLE test_table ON CLUSTER my_cluster 
``` 

**Expected behavior**
Truncate doesnt block shard, table clean successfully.

**Error message and/or stacktrace**
1st instance of shard: 
```
2019.08.29 11:23:21.890845 [ 3 ] {5d8fd170-12c9-4138-b930-ba8f3529a84f} <Debug> executeQuery: (from 0.0.0.0:0, user: ) /* ddl_entry=query-0000000122 */ TRUNCATE TABLE test_table
2019.08.29 11:23:21.894860 [ 16 ] {} <Debug> test_table (ReplicatedMergeTreeQueue): Pulled 1 entries to queue.
2019.08.29 11:23:21.896648 [ 39 ] {} <Trace> test_table: Renaming temporary part tmp_fetch_1567036800_26_26_0 to 1567036800_26_26_0.
2019.08.29 11:23:21.902978 [ 53 ] {} <Trace> InterserverIOHTTPHandler-factory: HTTP Request for InterserverIOHTTPHandler-factory. Method: POST, Address: localhost:46080, User-Agent: none, Content Type: , Transfer Encoding: identity
2019.08.29 11:23:21.903035 [ 53 ] {} <Trace> InterserverIOHTTPHandler: Request URI: /?endpoint=DataPartsExchange%3A%2Fclickhouse%2Ftables%2Ftest%2F02%2Ftest_table%2Freplicas%2Fclickhouse-02&part=1567036800_27_27_0&compress=false
2019.08.29 11:23:21.903116 [ 53 ] {} <Trace> test_table (Replicated PartsService): Sending part 1567036800_27_27_0
2019.08.29 11:23:21.904635 [ 39 ] {} <Debug> test_table: Fetched part 1567036800_26_26_0 from /clickhouse/tables/test/02/test_table/replicas/clickhouse-03
2019.08.29 11:23:21.906140 [ 29 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 1567036800_18_18_0, total 1 rows starting from the beginning of the part, columns: field
2019.08.29 11:23:21.906399 [ 29 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 1567036800_19_19_0, total 1 rows starting from the beginning of the part, columns: field
2019.08.29 11:23:21.906647 [ 29 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 1567036800_20_20_0, total 1 rows starting from the beginning of the part, columns: field
2019.08.29 11:23:21.906899 [ 29 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 1567036800_21_21_0, total 1 rows starting from the beginning of the part, columns: field
2019.08.29 11:23:21.907173 [ 29 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 1567036800_22_22_0, total 1 rows starting from the beginning of the part, columns: field
2019.08.29 11:23:21.907419 [ 29 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 1567036800_23_23_0, total 1 rows starting from the beginning of the part, columns: field
2019.08.29 11:23:21.907741 [ 53 ] {} <Information> InterserverIOHTTPHandler: Done processing query
```

2nd instance of shard: 
```
2019.08.29 11:27:21.730966 [ 3 ] {46272390-bcac-45b4-b447-8687983ed165} <Debug> DDLWorker: Processing task query-0000000124 (TRUNCATE TABLE test ON CLUSTER my_cluster)
2019.08.29 11:27:21.732567 [ 3 ] {46272390-bcac-45b4-b447-8687983ed165} <Debug> DDLWorker: Executing query: TRUNCATE TABLE test
2019.08.29 11:27:32.719626 [ 3 ] {46272390-bcac-45b4-b447-8687983ed165} <Debug> DDLWorker: Waiting a watch
```

**Additional context**
During `SELECT * FROM system.processes` i see that my query with truncate is stuck and cant be killed. 

UPD: is this issue already fixed here ? https://github.com/yandex/ClickHouse/pull/6695

**Comment by alexey-milovidov** [comment]
Deadlock on TRUNCATE of Replicated tables was fixed in #7974 version 19.14.3.3.

**Comment by alexey-milovidov** [comment]
> UPD: is this issue already fixed here ? #6695

Yes.
